### DISCUSSION TITLE
Option for default keymaps

### DISCUSSION THREAD

**Discussion by radvil** [LazyVim/LazyVim#1051] [Ideas]
Hi @folke, I think it is a good idea if we have an option to silent all of the default keymap like this, because I found it a bit distracting whenever I hit any builtin keymap the noice prompts kinda flashing, I use the default lazyvim keymap, but later I need to copy pasting all of those and defnining them myself in my `config/keymaps.lua`. Mine looks like this now..

```lua
local function map(mode, lhs, rhs, opts)
  local keys = require("lazy.core.handler").handlers.keys
  ---@cast keys LazyKeysHandler
  -- do not create the keymap if a lazy keys handler exists
  if not keys.active[keys.parse({ lhs, mode = mode }).id] then
    opts = opts or {}
    opts.silent = require("lazyvim.util").has("noice") or vim.opt.cmdheight == 0 or opts.silent ~= false
    if opts.remap and not vim.g.vscode then
      opts.remap = nil
    end
    vim.keymap.set(mode, lhs, rhs, opts)
  end
end
```

**Comment by folke** [comment]
I have the below in my personal config, but I'm a bit hesitant to add that to LazyVim.

You should add this before calling lazy:

```lua
-- make all keymaps silent by default
local keymap_set = vim.keymap.set
---@diagnostic disable-next-line: duplicate-set-field
vim.keymap.set = function(mode, lhs, rhs, opts)
  opts = opts or {}
  opts.silent = opts.silent ~= false
  return keymap_set(mode, lhs, rhs, opts)
end

**Reply by radvil** [reply]
That would work, I think it is good to have that on noice doc as a recipe or tips since many beginner like me start to use noice without knowing how it works under the hood, unless we dive into the code ourself
